commit_rmn_ocb/types.go

```diff
@@ -0,0 +1,348 @@
+			aggObs.GasPrices[gasPriceChain.ChainSel] =
+				append(aggObs.GasPrices[gasPriceChain.ChainSel], gasPriceChain.GasPrice)
+		}
+
+		// TokenPrices
+		for _, tokenPrice := range obs.TokenPrices {
+			aggObs.TokenPrices[tokenPrice.TokenID] =
+				append(aggObs.TokenPrices[tokenPrice.TokenID], tokenPrice.Price)
+		}
+
+		// OnRampMaxSeqNums
+		for _, seqNumChain := range obs.OnRampMaxSeqNums {
+			aggObs.OnRampMaxSeqNums[seqNumChain.ChainSel] =
+				append(aggObs.OnRampMaxSeqNums[seqNumChain.ChainSel], seqNumChain.SeqNum)
+		}
+
+		// OffRampMaxSeqNums
+		for _, seqNumChain := range obs.OffRampMaxSeqNums {
+			aggObs.OffRampMaxSeqNums[seqNumChain.ChainSel] =
+				append(aggObs.OffRampMaxSeqNums[seqNumChain.ChainSel], seqNumChain.SeqNum)
+		}
+
+		// FChain
+		for chainSel, f := range obs.FChain {
+			aggObs.FChain[chainSel] = append(aggObs.FChain[chainSel], f)
+		}
+	}
+
+	return aggObs
+}
+
+// ConsensusObservation holds the consensus values for all chains across all observations in a round
+type ConsensusObservation struct {
+	// A map from chain selectors to each chain's consensus merkle root
+	MerkleRoots map[cciptypes.ChainSelector]MerkleRoot
+
+	// A map from chain selectors to each chain's consensus gas prices
+	GasPrices map[cciptypes.ChainSelector]cciptypes.BigInt
+
+	// A map from token IDs to each token's consensus price
+	TokenPrices map[types.Account]cciptypes.BigInt
+
+	// A map from chain selectors to each chain's consensus OnRamp max sequence number
+	OnRampMaxSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus OffRamp max sequence number
+	OffRampMaxSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus f (failure tolerance)
+	FChain map[cciptypes.ChainSelector]int
+}
+
+// GasPricesSortedArray returns a sorted list of gas prices
+func (co ConsensusObservation) GasPricesSortedArray() []cciptypes.GasPriceChain {
+	gasPrices := make([]cciptypes.GasPriceChain, 0, len(co.GasPrices))
+	for chain, gasPrice := range co.GasPrices {
+		gasPrices = append(gasPrices, cciptypes.NewGasPriceChain(gasPrice.Int, chain))
+	}
+
+	sort.Slice(gasPrices, func(i, j int) bool {
+		return gasPrices[i].ChainSel < gasPrices[j].ChainSel
+	})
+
+	return gasPrices
+}
+
+// TokenPricesSortedArray returns a sorted list of token prices
+func (co ConsensusObservation) TokenPricesSortedArray() []cciptypes.TokenPrice {
+	tokenPrices := make([]cciptypes.TokenPrice, 0, len(co.TokenPrices))
+	for tokenID, tokenPrice := range co.TokenPrices {
+		tokenPrices = append(tokenPrices, cciptypes.NewTokenPrice(tokenID, tokenPrice.Int))
+	}
+
+	sort.Slice(tokenPrices, func(i, j int) bool {
+		return tokenPrices[i].TokenID < tokenPrices[j].TokenID
+	})
+
+	return tokenPrices
+}
+
+type CommitPluginOutcomeType int
+
+const (
+	ReportIntervalsSelected CommitPluginOutcomeType = iota
+	ReportGenerated
+	ReportEmpty
+	ReportNotYetTransmitted
+	ReportTransmitted
```

Comment:
These three:

* `ReportNotYetTransmitted`
* `ReportTransmitted`
* `ReportNotTransmitted`

seem to be so lexicographically similar I feel like its easy to mistake one for the other. How about changing `ReportNotYetTransmitted` to `ReportInFlight`?

Reply:
Updated